akka-actor-typed/src/main/scala/akka/actor/typed/receptionist/Receptionist.scala

```diff
@@ -200,6 +200,79 @@ object Receptionist extends ExtensionId[Receptionist] {
   def registered[T](key: ServiceKey[T], serviceInstance: ActorRef[T]): Registered =
     Registered(key, serviceInstance)
 
+  /**
+   * Remove association between the given [[akka.actor.typed.ActorRef]] and the given [[ServiceKey]].
+   *
+   * De-registration will be acknowledged with the [[Unregistered]] message to the given replyTo actor
+   * if there is one.
```

Comment:
Add a caveat here about the unregistration being eventually consistent meaning that there is no guarantee the service actor will not get more messages after it saw the `Unregistered` message, it only means the local receptionist saw and acted on the command.